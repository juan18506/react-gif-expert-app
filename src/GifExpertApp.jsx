import { useState } from 'react';

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([ 'Action', 'Drama' ]);

  const onAddCategory = () => setCategories([ ...categories, 'Fantasy' ]);

  return (
    <>
      <h1>GifExpertApp</h1>

      <button onClick={ onAddCategory }>Add Category</button>

      <ol>
        { 
          categories.map( category => (
            <li key={ category }>{ category }</li>
          ))
        }
      </ol>
    </>
  );
}
