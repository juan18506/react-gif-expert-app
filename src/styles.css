* {
  font-family: Helvetica, Arial, sans-serif;
  background-color: rgb(234, 234, 234);
}

body {
  padding: 3.75rem;
}

#root,
section {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  gap: 1em;
}

form {
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}

input {
  width: 80%;
  padding: .625rem .9375rem;
  background-color: white;
  border-radius: .3125rem;
  border: 1px solid rgb(97, 32, 158);
  color: black;
  font-size: 1.2rem;
  outline: none;
}

h1 {
  font-size: 3rem;
}

h2 {
  font-size: 1.5rem;
}

h3 {
  font-size: 3rem;
  margin-bottom: .3125rem;
}

.card-grid {
  padding: 0;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  gap: 1em;
}

.card {
  height: 0%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-content: center;
  align-items: center;
  border-radius: .625rem;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
  overflow: hidden;
  background-color: white;
}

.card p {
  padding: .3125rem 1.25rem 0px 1.25rem;
  margin-top: .3125rem;
  flex: 1;
  text-align: center;
  font-size: 1.5rem;
  background-color: white;
}

.card img {
  width: 100%;
}
