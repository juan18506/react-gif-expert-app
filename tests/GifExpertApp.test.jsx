import { describe, test, it, expect } from 'vitest';
import { screen, render, fireEvent } from '@testing-library/react';
import { GifExpertApp } from '../src/GifExpertApp';

describe('Tests in <GifExpertApp />', () => {
  const category = 'drama';

  it('should match with snapshot', () => {
    const { container } = render( <GifExpertApp /> );
    expect( container ).toMatchSnapshot();
  });

  it('should show title and should not show category at start', () => {
    render( <GifExpertApp /> );
    expect( screen.getByText('GifExpertApp') ).toBeTruthy();
    expect( screen.queryByText(category) ).toBeFalsy();
  });

  test('updateCategories should change categories', () => {
    render( <GifExpertApp /> );
    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    fireEvent.input( input, { target: { value: category } } );
    fireEvent.submit( form );

    expect( screen.getByText(category) ).toBeTruthy();
  });

  test('updateCategories should not change categories if category is repeated', () => {
    render( <GifExpertApp /> );
    const input = screen.getByRole('textbox');
    const form = screen.getByRole('form');

    fireEvent.input( input, { target: { value: category } } );
    fireEvent.submit( form );

    fireEvent.input( input, { target: { value: category } } );
    fireEvent.submit( form );

    expect( screen.getAllByText(category).length ).toBe( 1 );
  });
});
