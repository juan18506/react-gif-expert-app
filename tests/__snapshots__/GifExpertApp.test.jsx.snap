// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests in <GifExpertApp /> > should match with snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Search gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
